engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -726,16 +778,40 @@ namespace dmGameObject
 
         dmGameObject::PropertyOptions property_options;
         property_options.m_Index = 0;
+        property_options.m_HasKey = 0;
 
         // Options table
         if (lua_gettop(L) > 3)
         {
             luaL_checktype(L, 4, LUA_TTABLE);
             lua_pushvalue(L, 4);
 
+            lua_getfield(L, -1, "key");
+            if (!lua_isnil(L, -1))
+            {
+                if (lua_isstring(L, -1))
+                {
+                    property_options.m_Key = dmHashString64(lua_tostring(L, -1));
+                }
+                else if (dmScript::IsHash(L, -1))
+                {
+                    property_options.m_Key = dmScript::CheckHash(L, -1);
+                }
+                else
+                {
+                    return luaL_error(L, "Invalid key argument in options table. It should be string or hash.");
+                }
+                property_options.m_HasKey = 1;
+            }
+            lua_pop(L, 1);
+
             lua_getfield(L, -1, "index");
             if (!lua_isnil(L, -1)) // make it optional
-            {
+            {   
+                if (property_options.m_HasKey)
+                {
+                    return luaL_error(L, "Options table should contain either key or index value.");
```

Comment:
I think the message is a bit ambiguous.
Perhaps "Options table cannot contain both 'key' and 'index'" ?

Reply:
👍